[PrishaVP] iP

PrishaVP opened the pull request:
# MeowMeow Chatbot 🐱

> "MeowMeow helps you manage your tasks efficiently, so you can focus on more important things... like chasing laser pointers!" – _Anonymous Cat_

MeowMeow is:

- text-based
- user-friendly
- ~~FAST~~ *SUPER* FAST to use!

All you need to do is,

1. download it from [here](https://github.com/PrishaVP/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Task management
- [x] Deadline tracking
- [x] Event scheduling

If you are a Java programmer, you can use it to practice Java too. Here’s the `main` method:

```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

PrishaVP committed:
Level 0. Rename, Greet, Exit

PrishaVP committed:
Level 1. Echo

PrishaVP committed:
Level 2. Add, List

PrishaVP committed:
Level 3. Mark as Done

PrishaVP committed:
Level 4. ToDos, Events, Deadlines

PrishaVP committed:
Level 5. Handle Errors

PrishaVP committed:
Level 6. Delete

PrishaVP committed:
Duke A-TextUiTesting: Automated Text UI Testing

bbbbcccd commented on src/main/java/meowmeow.java:
```diff
@@ -0,0 +1,94 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class meowmeow {
```
Will it be better for the class name to start with a captial letter?

bbbbcccd reviewed (commented):
Lgtm!

HanB1n commented on src/main/java/meowmeow.java:
```diff
@@ -1,3 +1,4 @@
+import java.util.LinkedList;
 import java.util.Scanner;
 public class meowmeow {
```
Should class names be renamed using CamelCase?

HanB1n commented on src/main/java/meowmeow.java:
```diff
@@ -46,24 +46,32 @@ public static void main(String[] args) {
+                }
             } else if (input.startsWith("event ")) {
                 // Add an Event task
                 String[] parts = input.substring(6).split(" /from | /to ");
```
Would it be better if there was error handling in this steps?
Perhaps using try and catch blocks would help with possible Index Errors?

HanB1n reviewed (commented):
The code was generally well written and easy to understand.
A general suggestion: Using switch case might be good to improve code readability and reduce the amount of if else statements.

PrishaVP committed:
Add save functionality

PrishaVP committed:
Add save functionality properly

PrishaVP committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add save functionality properly
  Add save functionality

PrishaVP committed:
Add dates and times

PrishaVP committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add dates and times

PrishaVP committed:
Add OOP with Parser, Saving, Ui and TaskList classes

PrishaVP committed:
Delete files due to merge problem

PrishaVP committed:
Add more OOP with Parser, Saving, TaskList and UI classes

PrishaVP committed:
Add more OOP with Parser, Saving, TaskList and UI classes

PrishaVP committed:
Create package meowmeow with all classes

PrishaVP committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

PrishaVP committed:
Add JUnit tests EventTest and SavingTest

PrishaVP committed:
Rename package

PrishaVP committed:
Add find function

PrishaVP committed:
Add JavaDoc headers for more than half of the non-private classes or methods

PrishaVP committed:
Fix most code convention problems

PrishaVP committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find function

PrishaVP committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc headers for more than half of the non-private classes or methods

PrishaVP committed:
Merge tag 'branch-A-CodingStandard'

* tag 'branch-A-CodingStandard':
  Fix most code convention problems

# Conflicts:
#	src/main/java/MeowMeow/Ui.java
#	src/main/java/meowmeow/Storage.java
#	src/main/java/meowmeow/Task.java

PrishaVP committed:
Fix Task class

PrishaVP committed:
Fix Storage class

PrishaVP committed:
Add practice GUI

PrishaVP committed:
Add practice GUI

PrishaVP committed:
Add GUI for MeowMeow

PrishaVP committed:
Update README.md

PrishaVP committed:
Merge branch 'master' of https://github.com/PrishaVP/ip

* 'master' of https://github.com/PrishaVP/ip:
  Update README.md

PrishaVP committed:
Improve code quality

Parser class is too long and handles the MeowMeow output too.
The parse method is too long and violates SLAP.
The Ui class is does not handle the MeowMeow output.
There are minor SLAP violations in other classes as well.

Refactoring the Parser class to only handle parsing while the Ui class
handles the MeowMeow output allows us to practice better OOP.
Fixing SLAP violations allows us to ensure good code quality so that
the code is readable.

Let's,
* remove all code handling the MeowMeow output from the Parser class
and put it into the Ui class.
* split up long methods into shorter methods
* correct poor code quality

This allows for our code to be more readable.

PrishaVP committed:
Add assertions to Storage and MainWindow classes

Storage class deals with loading to or saving from new or existing save
files, hence makes assumptions about whether the save file exists or
not or whether it is empty or has valid tasks in it.

MainWindow class requires non-null instances of MeowMeow and Parser.
It also takes in user input and takes in responses from MeowMeow.
It makes assumptions that these are not null.

These assumptions could indicate bugs in the code if violated.

Let's add assertions to the Storage and MainWindow classes.

This allows us to detect assertion failure and terminate the
execution if there is a possible bug.

PrishaVP committed:
Merge pull request #3 from PrishaVP/branch-A-Assertions

Add assertions to Storage and MainWindow classes

PrishaVP committed:
Merge branch 'master' into branch-A-CodeQuality

PrishaVP committed:
Merge pull request #2 from PrishaVP/branch-A-CodeQuality

Improve code quality

PrishaVP committed:
Merge branch 'master' of https://github.com/PrishaVP/ip

* 'master' of https://github.com/PrishaVP/ip:
  Add assertions to Storage and MainWindow classes

PrishaVP committed:
Merge branch 'master' of https://github.com/PrishaVP/ip

* 'master' of https://github.com/PrishaVP/ip:
  Improve code quality

PrishaVP committed:
Merge branch 'branch-A-CodeQuality' remote into branch-A-CodeQuality

* 'branch-A-CodeQuality' of https://github.com/PrishaVP/ip:
  Add assertions to Storage and MainWindow classes

PrishaVP committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:

Fix merge problem

PrishaVP committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:

PrishaVP committed:
Add extension Do Within Period Tasks

PrishaVP committed:
Improve GUI

PrishaVP committed:
Add personality through name, responses and GUI

PrishaVP committed:
Improve JavaDoc using AI guidance

PrishaVP committed:
Add credits

PrishaVP committed:
Update README.md

PrishaVP committed:
Update README.md

PrishaVP committed:
Update README.md

PrishaVP committed:
Update README.md

PrishaVP committed:
Merge branch 'master' of https://github.com/PrishaVP/ip

* 'master' of https://github.com/PrishaVP/ip:
  Update README.md
  Update README.md
  Update README.md
  Update README.md

PrishaVP committed:
Add Ui.png to docs folder

PrishaVP committed:
Update README.md

PrishaVP committed:
Merge branch 'master' of https://github.com/PrishaVP/ip

* 'master' of https://github.com/PrishaVP/ip:
  Update README.md

PrishaVP committed:
Update Storage tests

PrishaVP committed:
Update README.md

PrishaVP committed:
Update README.md

PrishaVP committed:
Update README.md

PrishaVP committed:
Merge branch 'master' of https://github.com/PrishaVP/ip

* 'master' of https://github.com/PrishaVP/ip:
  Update README.md

PrishaVP committed:
Add more tests
